lib/core/sessions.js

```diff
@@ -125,7 +135,7 @@ class ClientSession extends EventEmitter {
 
     // release the server session back to the pool
     this.sessionPool.release(this.serverSession);
-    this.serverSession = null;
+    this[kServerSession] = undefined;
```

Comment:
I generally prefer to use `delete` rather than setting a property to undefined. Is there any reason to not use it, or is it just a matter of preference?

Reply:
I'm not sure how relevant this is in 2020, but the short answer is that `delete` can cause v8 to modify the "hidden class" backing a variable such that you introduce performance deoptimization (see more [here](https://v8.dev/blog/fast-properties)). We prefer to avoid using `delete` in the codebase, in particular in objects that often live in hot paths like sessions.